### PR TITLE
Create React Native Windows version variable in new app and lib vcxproj files

### PR DESCRIPTION
**Pull request by Yajur-Grover**
### Type of Change
_Erase all that don't apply._
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

### Why
What is the motivation for this change? Add a few sentences describing the context and overall goals of the pull request's commits.

Resolves [Add Relevant Issue Here]

### What
What changes were made to the codebase to solve the bug, add the functionality, etc. that you specified above.

## Screenshots
Add any relevant screen captures here from before or after your changes. 

## Testing
If you added tests that prove your changes are effective or that your feature works, add a few sentences here detailing the added test scenarios.

_Optional_: Describe the tests that you ran locally to verify your changes.

## Changelog
Should this change be included in the release notes: _indicate yes or no_

Add a brief summary of the change to use in the release notes for the next release.
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/13972)

### CODE REVIEW

File: vnext/templates/cpp-app/windows/MyApp/MyApp.vcxproj [hunk 1]
```diff
@@ -13,6 +13,7 @@
     <DefaultLanguage>en-US</DefaultLanguage>
     <MinimumVisualStudioVersion>17.0</MinimumVisualStudioVersion>
     <AppxPackage>false</AppxPackage>
+    <ReactNativeWindowsVersion>{{ rnwVersion }}</ReactNativeWindowsVersion>
```
Comment (acoates-ms) [inline vnext/templates/cpp-app/windows/MyApp/MyApp.vcxproj:16]: I beleive that this variable could contain things like prerelease.1.  Which wouldn't work with https://learn.microsoft.com/en-us/visualstudio/msbuild/property-functions?view=vs-2022#msbuild-version-comparison-functions.
Maybe we should stamp a variable with a version that never includes the prerelease part of the build number.  And maybe another variable which includes the full version including the prerelease.

Comment (Yajur-Grover) [reply vnext/templates/cpp-app/windows/MyApp/MyApp.vcxproj:16]: This would also apply to canary versions correct?

Comment (jonthysell) [reply vnext/templates/cpp-app/windows/MyApp/MyApp.vcxproj:16]: What is the purpose of this> The property for version of RNW being used is already defined in `vnext/PropertySheets/Generated/PackageVersion.g.props`

**Comment by Yajur-Grover** [comment]
Closing as not needed.

**Pull request closed without merging**